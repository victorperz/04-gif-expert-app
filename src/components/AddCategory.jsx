import { useState } from 'react'

export const AddCategory = ({ onNewCategory }) => {
    // console.log({setCategories});
    const [inputValue, setInputValue] = useState('')

    // const onInputChange = (event) => {
    //     // console.log(event.target.value);
    //     setInputValue(event.target.value);
    // }

    const onInputChange = ({target}) => {
        setInputValue(target.value);
    }

    const onSubmit = (event) => {
        event.preventDefault();
        if( inputValue.trim().length <= 1) return;

        // setCategories( (categories) =>[ inputValue , ...categories]);
        onNewCategory(inputValue.trim() );
        setInputValue('');
    }

  return (
    <form onSubmit={ (event) => onSubmit(event) }>

        <input 
            type="text"
            placeholder="Buscar Gift"
            value={ inputValue }
            // onChange={ (event) =>  onInputChange(event) }
            onChange={ onInputChange }
        />

    </form>
    
  )
}
