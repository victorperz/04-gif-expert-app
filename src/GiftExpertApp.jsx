import { useState } from 'react'
import { AddCategory, GifGrid } from './components';
// import { GifGrid } from './components/GifGrid';


export const GiftExpertApp = () => {
    
    const [categories, setCategories] = useState(['Kimetsu no yaiba' ]);
    // console.log(categories);

    const onAddCategory = ( newCategory) => {

        if ( categories.includes(newCategory) ) return;

        setCategories([ newCategory, ...categories]);
    }

    return (
    <>

       <h1>GiftExpertApp</h1>


       <AddCategory 
                // setCategories={ setCategories } 
                onNewCategory={(value) => onAddCategory(value)}
        />

            { 
              categories.map( (category) => (
                    <GifGrid 
                        key={ category } 
                        category={ category }
                    />
               ))
            }


    </>
  )
}
