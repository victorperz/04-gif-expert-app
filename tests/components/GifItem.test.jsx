import { render , screen} from '@testing-library/react';
import { GifItem } from '../../src/components';


describe('Pruebas del componente GifItem', () => { 
    
    const title = 'Kimetsu no yaiba';
    const url   = `https://kimetsu-no-yaiba.com/kimetsu.jpg`;

    test('Evaluar el screenshop', () => { 
        const { container } = render( <GifItem title = {title} url = {url}/>);
        expect( container ).toMatchSnapshot();
     });

     test('debe de mostrar la imagen con el URL y el ALT indicado', () => { 
        
        render( <GifItem title = {title} url = {url}/>);
        // screen.debug();

        // expect( screen.getByRole('img').src).toBe( url );
        // console.log(screen.getByRole('img').src);

        const {src, alt} = screen.getByRole('img');
        expect( src ).toBe( url );
        expect( alt ).toBe( alt );

      });

      test('debe de mostrar el titulo en el componente', () => { 
        
        render( <GifItem title={ title } url={ url } /> );
        expect( screen.getByText( title ) ).toBeTruthy();

       });
 });