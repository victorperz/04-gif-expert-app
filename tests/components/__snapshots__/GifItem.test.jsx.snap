// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del componente GifItem Evaluar el screenshop 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Kimetsu no yaiba"
      src="https://kimetsu-no-yaiba.com/kimetsu.jpg"
    />
    <p>
      Kimetsu no yaiba
    </p>
  </div>
</div>
`;
